### ISSUE TITLE
[Bug]: Send and swap flow for unsupported networks

### ISSUE THREAD

**Issue by sleepytanya** [MetaMask/metamask-extension#25107]
### Describe the bug

  In the context of the send+swap flow on unsupported networks, considering the substantial user base operating within these networks, introducing a notification system or potentially disabling this feature for networks beyond the mainnet and BNB might significantly improve the user experience. 
      Currently, when users on unsupported networks attempt to use the send+swap feature, they are allowed to select a destination token, only to encounter 'No quotes available' for the native token and 'Error fetching quotes' for other tokens.
       A more user-friendly approach could involve defaulting to the regular send flow for non-supported networks (removing the option to select a destination token). 

### Expected behavior

-

### Screenshots/Recordings

<img width="366" alt="Screenshot 2024-06-05 at 23 39 13" src="https://github.com/MetaMask/metamask-extension/assets/104780023/d6eedd4c-1cb5-437c-98df-82a43e3089e0">

<img width="371" alt="Screenshot 2024-06-05 at 23 40 47" src="https://github.com/MetaMask/metamask-extension/assets/104780023/5bd097ba-7e1d-4a77-91ca-b60f0bc51466">

### Steps to reproduce

1. Connect to unsupported network (e.g., Polygon, Arbitrum, etc.)
2. Start send+swap flow
3. Select native token as an origin token and any other token for the destination
4. See 'No quotes available' message
5. Select non-native token as an origin
6. Select any token as a destination token
7.  'Error fetching quotes' is shown

### Version

develop branch

### Build type

None

### Browser

Chrome

### Operating system

MacOS

**Comment by BZahory** [comment]
This was addressed in a meeting yesterday: https://docs.google.com/document/d/1bfuJ3oRx3KHBtCZnySJjuTJ8r4FVVqDV3Eb-xxYJ4Zs/edit#heading=h.4mo58br3ahpy

tl;dr: this is expected since the API isn't updated yet, but we expect it to be updated by mid-next week. Keeping this issue open until this actually happens

**Comment by sleepytanya** [comment]
Currently for unsupported networks we allow users to select destination token and then display 'No quotes available' or 'Error fetching quotes' message.


https://github.com/MetaMask/metamask-extension/assets/104780023/f11298b6-363b-4c09-b936-992c567452b7

**Comment by danjm** [comment]
@BZahory Why not disable the dropdown if the network is unavailable?

**Comment by BZahory** [comment]
> @BZahory Why not disable the dropdown if the network is unavailable?

That's the current behavior. We previously assumed that all swap networks would be supported. Exact swap+send networks are available here: https://swap.dev-api.cx.metamask.io/featureFlags; I'll make a PR to use these

**Comment by sleepytanya** [comment]
The issue is present in the latest [v12.0.0 build]([https://github.com/MetaMask/metamask-extension/commit/d9c4f7514532540a2567bbc65802af3797c5ec5c]) @BZahory 

<img width="813" alt="Screenshot 2024-06-29 at 17 03 52" src="https://github.com/MetaMask/metamask-extension/assets/104780023/0d812a8d-8e87-4e64-910e-00340e06de2f">

**Comment by sleepytanya** [comment]
The issue is present in the v12.0.0. beta:

<img width="486" alt="Screenshot 2024-07-15 at 22 04 31" src="https://github.com/user-attachments/assets/b13d89e2-0a7b-4cb0-8e6a-f9d9a87ea04d">

<img width="491" alt="Screenshot 2024-07-15 at 22 06 13" src="https://github.com/user-attachments/assets/16046376-64d7-4e85-a652-dc4bf8b1e626">

**Comment by sleepytanya** [comment]
v12.0.0 branch with `SWAPS_USE_DEV_APIS=true`

Send+swap is not disabled on Avalanche and Linea, other popular networks seem to work as expected.

<img width="362" alt="avalanceNoQoutes" src="https://github.com/user-attachments/assets/99d48513-5989-48de-adf2-32999e650938">

<img width="369" alt="lineaNoQuotes" src="https://github.com/user-attachments/assets/b98abf61-aaf5-48a5-9da7-27e457fdad65">

**Comment by BZahory** [comment]
For Linea, another route is supported. Avalanche's unavailability is related to an API bug which @callumelvidge is investigating.

#### Linea
<img width="442" alt="Screenshot 2024-07-17 at 4 05 03 PM" src="https://github.com/user-attachments/assets/2b576ca7-94c2-45a7-aa3a-9332ccb93117">

**Comment by sleepytanya** [comment]
Production v12.1.0

Linea, Base and Optimism chains are not supported:

<img width="1557" alt="Screenshot 2024-08-09 at 18 10 32" src="https://github.com/user-attachments/assets/4c5cd303-bb33-4665-acbc-41c47212869e">

**Comment by sleepytanya** [comment]
Beta 12.1.0 - Linea and Base:


<img width="405" alt="Screenshot 2024-08-23 at 17 32 28" src="https://github.com/user-attachments/assets/36ceeaa8-bdff-457f-a910-80d26ace33f5">


<img width="411" alt="Screenshot 2024-08-23 at 17 38 21" src="https://github.com/user-attachments/assets/dc7b1af3-b151-4a3f-a9a9-b499cee1fada">

**Comment by sleepytanya** [comment]
RC 12.3.0 - 'No quotes' on Linea Mainnet:


https://github.com/user-attachments/assets/f82c4be2-692a-4b4e-a59c-6e19bf6ae387

**Comment by sleepytanya** [comment]
Present in the RC 12.8.0.

Quotes are not disabled on two unsupported networks, 'No quotes available' messages is displayed - Linea and zkSync Era:

<img width="1493" alt="Screenshot 2024-11-22 at 00 41 02" src="https://github.com/user-attachments/assets/8f65ebb0-e7fa-498d-809a-d63c9c713a0f">

<img width="1487" alt="Screenshot 2024-11-22 at 00 41 34" src="https://github.com/user-attachments/assets/61577f73-5e2c-4c7c-b1f1-ec7a2ac421ef">

<img width="1439" alt="Screenshot 2024-11-22 at 00 37 19" src="https://github.com/user-attachments/assets/dfb212f3-cef7-4cc1-9912-eeb0da4e6972">

**Comment by sleepytanya** [comment]
Present in 12.11.0:


<img width="1275" alt="Image" src="https://github.com/user-attachments/assets/adb3e807-ee28-4189-bed0-707fad6377c1" />

<img width="1170" alt="Image" src="https://github.com/user-attachments/assets/514a9539-e883-4cab-a350-be21cf55017e" />




<img width="1161" alt="Image" src="https://github.com/user-attachments/assets/cbca0ffe-4298-4b86-80f4-614f29d57b7a" />

**Comment by dylanbutler1** [comment]
Some of these seem to be simply that the value is too low for there to be any valid routes. I tested a bunch of these and got back routes for coins that have significant liquidity. For those that don't, it would seem this is the expected behavior. Please let me know if we can close this one out @sleepytanya

**Comment by sleepytanya** [comment]
@dylanbutler1 

Thank you for looking into this!
Are we talking about the same flow - send+swap? For me only wrapping works on Linea, everything else is 'No quotes available' with  the error 'not supported on this network'. On zkSync I couldn't find a working pair.
12.15.0, Linea:



https://github.com/user-attachments/assets/13b48ade-3bbf-44e6-a97a-a837623db48a


12.15.0, zkSync:


<img width="1351" alt="Image" src="https://github.com/user-attachments/assets/26d7f6a2-6579-4acd-8072-ba355474ba6d" />

**Comment by dylanbutler1** [comment]
@sleepytanya interesting, I must've been testing Swaps. Sorry about that! We are looking into it.

**Comment by dylanbutler1** [comment]
I've disabled Swap & Send on zksync and linea, it's already disabled for arbitrum.
